### DISCUSSION TITLE
How does WSL manage the /tmp folder cleanup?

### DISCUSSION THREAD

**Discussion by tigerinus** [microsoft/WSL#7758] [Q&A] [answered]
In Ubuntu there is /etc/tmpfiles.d for tmp clean up settings. A service would clean up the /tmp folder periodically.

Since there is no systemd or init.d on WSL2, how does it do the clean up for long term?

(My /tmp folder is like 20G+ now)

**Comment by craigloewen-msft** [comment, marked as answer]
WSL has a custom init process, and we don't do anything special for the /tmp folder. At some point if we adopted systemd we could inherit its handling of the /tmp folder!

**Reply by tigerinus** [reply]
adopting systemd would mean either:

- [memory reclaim](https://devblogs.microsoft.com/commandline/memory-reclaim-in-the-windows-subsystem-for-linux-2/) is no longer applicable.
- wsl2 gets a slower startup

I'm thinking about a separate scheduled job service that has access to WSL volume, responsible for periodical jobs like cleanup, reporting status, etc. It's probably a Windows service at host level...

**Reply by Biswa96** [reply]
Before the actual init process, another init ELF binary in initrd.img file is executed. It does some those steps which you mentioned, not the actual init process.

**Comment by matthewpersico** [comment]
* Create a shortcut in %AppData%\Microsoft\Windows\Start
Menu\Programs\Startup with the following command:

C:\Windows\System32\wsl.exe sudo /etc/init.d/cron start

Now schedule your /tmp clean job via root cron. Make it read a status file
of the last time you cleaned /tmp and if that time + some interval < now,
do the cleanup. Or experiment to see if @reboot applies when this job
starts each time you restart your PC.

On Tue, Dec 7, 2021 at 3:20 PM Biswapriyo Nath ***@***.***>
wrote:

> Before the actual init process, another init ELF binary in initrd.img file
> is executed. It does some those steps which you mentioned, not the actual
> init process.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/WSL/discussions/7758#discussioncomment-1768595>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAX365TY5V3UGBH5LCZDIUDUPZUAPANCNFSM5IZ2LWAQ>
> .
>


-- 
Matthew O. Persico
